### PR TITLE
Improved the accuracy of the `FPSCounter` by using Flutter's internal frame timings.

### PR DESCRIPTION
**Pull request by wolfenrain**
# Description

Changed the way the FPS gets calculated, instead of using the delta values from the update loop I am using `WidgetsBinding.instance.addTimingsCallback`, Flutter calls this with a list of `FrameTiming` which we can use to calculate the FPS value. 

This does mean that it is no longer required to return `true` in `recordFPS` because the values will always get recorded by Flutter. It also means that `recordDt()` will no longer have any purpose, but I kept it in, in case someone is using it for something else. 

Quite from the Discord:
> Well I do think it is more accurate. For instance in my 3D game I noticed the flame FPS sometimes dropped to 29.90 and then back to 60 within a single frame. Which is strange for that to happen in a single frame. With my variant I can see it dropping to 48 and then 30 and then to 48 and back to 60 again(just a random example not actual numbers)
> 
> Spread over multiple frames as a FPS drop should be. Making me think that the flame FPS is kind off averaging out and approximating the value instead of being true to the Flutter render cycle.

**Notes**: I would recommend that this gets tested a bit more to see if my assumptions are correct.

## Type of change

Please delete options that are not relevant.

- [X] Improvement?
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `develop`
- [x] This PR is targeted to merge into `develop` (not `master`)
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [ ] I have made corresponding changes to the documentation
- [ ] I have added examples for new features in `doc/examples`
- [ ] The continuous integration (CI) is passing

### CODE REVIEW

**Review by spydon — APPROVED** [review]
Lgtm, just needs a documentation update.

File: lib/fps_counter.dart [hunk 1]
```diff
@@ -1,39 +1,33 @@
...
-    if (dts.isEmpty) {
-      return 0.0;
-    }
-    final double dtSum = dts.reduce((s, t) => s + t);
-    final double averageDt = dtSum / average;
-    return 1 / averageDt;
+    return _previousTimings.length *
+        _maxFrames /
+        _previousTimings.map((t) {
+          return (t.totalSpan.inMicroseconds ~/ frameInterval.inMicroseconds) +
```
[the remark this answers is not in the document]
Comment (wolfenrain) [reply lib/fps_counter.dart:29]: I had it as a one-liner but found that it loses in readability, so I changed it into this

**Comment by wolfenrain** [comment]
I still haven't had much time to fully test this myself nor update the docs, just wanted to give a headsup that I havent forgotten about it.

File: lib/fps_counter.dart [hunk 2]
```diff
@@ -1,39 +1,33 @@
...
+              1;
```
Comment (luanpotter) [inline lib/fps_counter.dart:30]: I think the multiline is fine but this +1 hopefully fits above? seems odd to have a +1 alone in a line

**Review by luanpotter — APPROVED** [review]
LGTM

**Pull request merged**